src/actions/actions.js

```diff
@@ -218,6 +242,36 @@ export function updateCellProperties(cellId, updatedProperties) {
   }
 }
 
+export function evaluateText() {
+  return (dispatch, getState) => {
+    const cm = window.ACTIVE_CODEMIRROR
+    const doc = cm.getDoc()
+
+    let actionObj
+
+    if (!doc.somethingSelected()) {
+      const { line } = doc.getCursor()
+      const [activeChunk] = getState()
+        .jsmdChunks
+        .filter(c => c.startLine <= line && line <= c.endLine)
+
+      actionObj = Object.assign({
```

Comment:
I think we'll be able to better unit this block & the unimplemented one below it.

Reply:
we spoke, decided we'll take this in a follow up